docs: update docs to reflect new PR targeting methods for release trains (#38401)
angular/angular: docs/CARETAKER.md

## Merging the PR

> A PR needs to have `PR action: merge` and `PR target: *` labels to be considered

Reviewer: We might need to adjust this here too. I now see a bit of inconsistency with `PR action` and `target`.

Author: I updated `PR target: *` to `target: *`, but I think we should check and be sure if we want `PR action` to be changed to `action` since `action` could theoretically apply to both Issues and PRs.

Reviewer: yeah, that's what I meant below. We should check if this is right.

Author: @jelbourn do you have an expectation here?

Reviewer 2: I think just `action: ` is fine. I also don't strictly see the need for the "action" prefix in the first place vs. labels like "ready to merge", "needs rebase", etc., but I don't particularly care about it that much.

Author: Updated docs throughout the repo to reflect the dropping of `PR` in `PR action`

I think there remains value in prefixing all of the actions with `action:` as it groups them together in the UI.

ready to merge. Merging is performed by running `ng-dev pr merge` with a PR number to merge.

> The tooling will automatically retrieve the PR and tests to confirm it merges into all targetted branches.

Reviewer: Nit: This seems to sound a bit wrong, unless I'm wrong.  should it be:

```
The tooling will automatically fetch a specified PR and perform tests to ensure it is ready for merge. If the PR
passes the tests, the tool will automatically merge it based on the applied target label.
```

> The tooling will automatically fetch a specified PR and perform tests to ensure it is ready for merge. If the PR

Reviewer 2: ```suggestion
The tooling automatically verifies that the given PR is ready for merge. If the PR
```

passes the tests, the tool will automatically merge it based on the applied target label.

To merge a PR run:

```
$ yarn ng-dev pr merge <pr number>
```

### Recovering from failed `merge-pr` due to conflicts

> The ng-dev pr merge script will automatically recover to the previous git state when the merge fails.

Reviewer 2: ```suggestion
The `ng-dev pr merge` tool automatically restores the previous git state when a merge fails.
```
